lms/static/js/courseware/courseware_factory.js

```diff
@@ -0,0 +1,25 @@
+;(function (define) {
+    'use strict';
+
+    define([
+            'jquery',
+            'logger',
+            'js/courseware/views/bookmarks_list_button'
+        ],
+        function($, Logger, BookmarksListButton) {
+            return function() {
+                // performs all actions common to all courseware. Currently this just includes
+                // adding an event to all link clicks.
+                $('a:not([href^="#"])').click(function(event) {
+                    Logger.log(
+                        "edx.ui.lms.link_clicked",
+                        {
+                            current_url: window.location.href,
+                            target_url: event.currentTarget.href
+                        });
+                });
+                return new BookmarksListButton();
```

Comment:
Nit: can you not return the button as the caller doesn't use it, and it is a bit confusing to understand why a CoursewareFactory would return a specific button. As we add more capabilities here it will just seem more confusing IMO. It then illustrates the strangeness of the fact that just the fact of constructing an instance of the button will have side effects (in this case, attaching event listeners to any buttons that have been added to the page). It might warrant a comment to explain this.